### DISCUSSION TITLE
Collection and global access gets out of sync in admin panel if the access depends on content

### DISCUSSION THREAD

**Discussion by joas8211** [payloadcms/payload#2191] [Feature Requests & Ideas]
# Bug Report

Payload version: 1.6.15

## Steps to Reproduce

1. Add following globals to your project.

```typescript
import { GlobalConfig } from "payload/types";

export const Settings: GlobalConfig = {
  slug: "settings",
  fields: [
    {
      type: "checkbox",
      name: "test",
      label: "Allow access to test global",
    },
  ],
};

export const Test: GlobalConfig = {
  slug: "test",
  fields: [],
  access: {
    read: async ({ req: { payload } }) => {
      const access = await payload.findGlobal({ slug: "settings" });
      return access.test;
    },
  },
};

```

2. Navigate to settings global.
3. Check "Allow access to test global" checkbox.
4. Observe that test global did not appear in globals list in side panel.
5. Refresh the page.
6. Observe that test global has now appeared.

## Expected behaviour

Test global should have appeared in step 4.

## Possible solution

Refresh access after a document is saved.

**Comment by joas8211** [comment]
This issue is related a problem in plugin I'm developing: https://github.com/joas8211/payload-tenancy/issues/2

**Comment by jmikrut** [comment]
Hey @joas8211 — I think the move here would be to open up a way to automatically re-fetch permissions, so that in your own custom components, you can re-request them at your discretion.

Handling this automatically every time a global or a collection document updates would make for quite a few unnecessary requests, so this is unlikely something that we will be able to solve internally within Payload, but a `fetchNewPermissions` method or similar from the `useAuth` hook could do the job for what you need.

This is more of a feature request though rather than an issue with Payload core, because as of now this is functioning as-intended. So I'll convert this to a discussion and tag it accordingly. However, I do think that the idea above, manually being able to re-fetch permissions, is pretty solid and we would happily accept a feature to build it. Not sure when the Payload team will be able to get to this, but i will keep it on our radar regardless.

What do you think? Would you be up for creating a PR for this? Otherwise, maybe you could wire in an automatic full page refresh when you need to refresh permissions?

**Reply by joas8211** [reply]
Sure, I could create a PR for this. I was actually trying to poke around the React hooks to get accesses to reload. So looked into the insides of useAuth hook already. If I'm remembering correctly there's already function for that, and as you already mentioned, it might be called `fetchNewPermissions`, but it's just not exposed via the context yet. It didn't occure to me that refreshing the page could work as a workaround.

What are the requirements for the PR to be accepted. I'm guessing docs needs to be updated in addition to the actual implementation? What else?

**Reply by joas8211** [reply]
Oh, I just found the contribution guidelines. My eyes just didn't see them at first. Seems to have all the info I need.
